modin/data_management/functions/groupby_function.py

```diff
@@ -120,13 +125,24 @@ def map(
         # right index and placing columns in the correct order.
         groupby_args["as_index"] = True
         groupby_args["observed"] = True
+        partition_selection = (
+            selection
+            if selection is None
+            else (
+                df.columns.intersection(selection)
+                if is_list_like(selection)
+                else selection
+            )
+        )
+
         if other is not None:
             # Other is a broadcasted partition that represents 'by' columns
             # Concatenate it with 'df' to group on its columns names
-            other = other.squeeze(axis=axis ^ 1)
+            if not drop:
+                other = other.squeeze(axis=axis ^ 1)
```

Comment:
if `drop` is False then the 'by' data came from the different frame and so have to be interpreted as an external Series here, otherwise, we should insert this as a column into the frame to get proper behavior